### ISSUE TITLE
Question: using custom ids

### ISSUE THREAD

**Issue by skc11** [wolf4ood/gremlin-rs#202]
Hi, is there a way to set our own Id property for vertices and edges when creating them ?

**Comment by andreclaudino** [comment]
I need the same feature. Tryied passing ~id property and upsert with coalesce (could not implement because I can't underst how to use coalesce), but look like it is not possible to use a custom id.

**Comment by Ech0riginal** [comment]
I do not believe TinkerPop3 has this ability out of the box. You can reference each Step [here](https://tinkerpop.apache.org/docs/current/reference/).

I did find that JanusGraph has the ability to store custom vertex IDs, here's the [relevant documentation](https://docs.janusgraph.org/advanced-topics/custom-vertex-id/).

**Comment by criminosis** [comment]
I've had success doing it to JanusGraph via an anonymous traversal like inside a coalesce  

> g.V("custom_vertex_id").fold().coalesce::<Vertex, _>[__.unfold(),
__.add_v("some_label_here")
 .property(gremlin_client::structure::T::Id, custom_vertex_id)]

But it doesn't seem like you can do on a traversal directly. If you try to do something like:
`graph.get_traversal().add_v("some_label_here").property(gremlin_client::structure::T::Id, custom_vertex_id)` you fight the type system.

 @wolf4ood  The `property()` [method there](https://github.com/wolf4ood/gremlin-rs/blob/d7ce0e42574458d7d5fef7c787684f3b74b88699/gremlin-client/src/process/traversal/graph_traversal.rs#L92) has a hard type requirement that key be `&str`, whereas the `property()` for an [anonymous traversal](https://github.com/wolf4ood/gremlin-rs/blob/d7ce0e42574458d7d5fef7c787684f3b74b88699/gremlin-client/src/process/traversal/builder.rs#L91-L93) (`__` returns a `TraversalBuilder`) takes a key of type `Into<GValue>`. Not sure if that's intentional?

**Comment by criminosis** [comment]
I took a stab at adding custom id support in my fork's branch. They're featured in this [PR](https://github.com/wolf4ood/gremlin-rs/pull/214) if anyone wants to try the branch until it gets merged & release.
